0010-zkapp-cli-lightnet-management.md

```diff
@@ -0,0 +1,100 @@
+# RFC for zkapp-cli enhancements to provide the local lightnet networks management
+
+Introduce new sub-commands to the existing zkapp CLI application to streamline the local lightnet networks management for development and testing purposes.
+
+## Summary
+
+We aim to extend the current zkapp CLI application by introducing the `lightnet` sub-command suite. This suite will centralize and simplify Docker containers and provided tools management tasks for local zkApps development and testing, enhancing user and developer experience.
+
+## Motivation
+
+The primary motivation is to simplify the local zkApps development and testing process by offering intuitive CLI commands.
+
+Expected Outcomes:
+
+- A streamlined workflow for managing local lightnet network [Docker containers](https://hub.docker.com/r/o1labs/mina-local-network) and tools provided by such containers.
+- Reduced complexity and better DX, eliminating manual 3rd party tools management.
+
+## Detailed design
+
+### zkapp CLI Interface Enhancements
+
+Under the primary command `zk`, the new `lightnet` sub-commands will be introduced:
+
+#### Help
+
+- **Command**: `zk lightnet`
+- **Output**: Help message with description of available sub-commands and their options.
+
+#### Start lightnet network
+
+- **Command**: `zk lightnet start <options>`
+- **Function**: Checks the env., starts the local lightnet Docker container with reasonable default configuration options and waits until the network sync. Repeating the command while network is up and running should not start a new container, but rather check the status of the existing one.
+- **Configuration options**:
+  - --mode: `single-node` or `multi-node` (default: `single-node`)
+  - --proof-level: `none` or `full` (default: `none`)
+  - --archive-node: `true` or `false` (default: `true`)
+  - --mina-branch: `rampup`, `berkeley` or `develop` (default: `rampup`)
+  - --dune-profile: `lightnet` or `devnet` (default: `lightnet`)
```

Comment:
When would someone use `devnet` for the dune profile? Is this something we need to expose? Are there invalid `mina-branch` and `dune-profile` combinations to look out for?

Reply:
> When would someone use devnet for the dune profile? 
In case when one will need the network properties closer to those in use for public networks. Use `multi-node` as network mode, `full` as proof level, `devnet` as Dune profile and you will get container that runs small network of predefined participants with constants and properties closer to the real world.

> Is this something we need to expose?
Yes, for fine tuning I think we need it.

> Are there invalid mina-branch and dune-profile combinations to look out for?
No, every major Mina branch that we publish [docker image for](https://hub.docker.com/r/o1labs/mina-local-network/tags) is built twice: with `lightnet` and `develop` Dune profiles. This results in 2 images per Mina branch. Image tags anatomy is also described in Docker Hub [repo description](https://hub.docker.com/r/o1labs/mina-local-network).

Do you think we need to document something in more details?